bot/utils/time.py

```diff
@@ -111,3 +111,33 @@ def parse_rfc1123(stamp: str) -> datetime.datetime:
 def format_infraction(timestamp: str) -> str:
     """Format an infraction timestamp to a more readable ISO 8601 format."""
     return dateutil.parser.isoparse(timestamp).strftime(INFRACTION_FORMAT)
+
+
+def get_duration_from_expiry(
+    expiry: str = None,
+    date_from: datetime.datetime = None,
+    max_units: int = 2
+) -> Optional[str]:
+    """
+    Returns a human-readable version of the  the duration between datetime.utcnow() and an expiry.
+
+    Unlike the original function, this function will force the precision to be 'seconds' by not passing it.
+    max_units specifies the maximum number of units of time to include (e.g. 1 may include days but not hours).
+    By default, max_units is 2
+
+    :param expiry: A string. If not passed in, will early return a None ( Permanent infraction ).
+    :param date_from: A datetime.datetime object. If not passed in, will use datetime.utcnow().
+    :param parts: An int, to show how many parts will be returned ( year - month or year - month - week - day ...).
+    """
+    if not expiry:
+        return None
+
+    date_from = date_from or datetime.datetime.utcnow()
+    date_to = dateutil.parser.isoparse(expiry).replace(tzinfo=None, microsecond=0)
+
+    expiry_formatted = format_infraction(expiry)
+
+    duration = humanize_delta(relativedelta(date_to, date_from), max_units=max_units)
+    duration_formatted = f" ({duration})" if duration else ''
+
+    return f"{expiry_formatted}{duration_formatted}"
```

Comment:
It's neither clear that this function returns a timestamp nor that it uses `format_infraction` specifically to format it. This functions seems to have been designed with only returning the duration in mind and then the timestamp was tacked on towards the end. It could only return the duration, *but* that would mean two function calls in the mod cogs. It may be fine to just rename this function to `format_infraction_with_duration` (preferably something shorter if you can think of it) and, of course, document that it returns a timestamp.

Reply:
You are right, I was just replacing `format_infraction` and avoid calling another command on top of it.
I've renamed the function to `format_infraction_with_duration` ( I can't think of any shorter name, `format_infraction_readable` kind of saying `format_infraction` is not readable, so I'd rather go with lengthier name ).